manager/src/main/java/com/redhat/service/bridge/manager/ProcessorServiceImpl.java

```diff
@@ -164,31 +165,33 @@ public Processor updateProcessorStatus(ProcessorDTO processorDTO) {
 
     @Transactional
     @Override
-    public Long getProcessorsCount(String bridgeId, String customerId) {
-        return processorDAO.countByBridgeIdAndCustomerId(bridgeId, customerId);
+    public Long getProcessorsCount(Bridge bridge) {
```

Comment:
I can see why of this change, but I think that having the methods not to depend on the full class but rather on the primitive key instance was intended. It lighten the API and makes it easier to test it for example.

Reply:
@lucamolteni Sorry I didn't get How it could makes easier to test it for example . Processors are not depends on customer id at all so passing customer id to processor methods are more confusing. 

There are multiple improvement that we could make here as following :
- Ideally we should not pass `Bridge` entity to other services at all. We should fetch `Bridge` entity inside `BridgeDAO` and then convert it to some `BridgeDTO` to pass bridge information internally.

- Inside `processorDAO.countByBridgeIdAndCustomerId` we are making a Query like `p.bridge.id=:bridgeId and p.bridge.customerId=:customerId`. With my changes as we already fetch Bridge instance using `bridgeId` & `customerId`, now we could make this call simple by just filter processors where `p.bridge.id=:bridgeId`.

Reply:
it's easier to test because you can call the API without having to have a fully constructed entity

Reply:
I think I'm +1 on this change. It feels more aligned with our other service interfaces where we are working with sub-resources. For example `ConnectorsService` takes in a `Processor` parameter to store the requested `Connector`.

I think this change, combined with the following, gives us a cleaner overall approach

- `ProcessorsAPI` is marked with `@Transactional` at the class-level to ensure we have a single DB transaction for the read of both the `Bridge` and the associated `Processors`
- The `BridgeServices.isReadyBridge(Bridge bridge)` call throws the `BridgeLifecycleException` if the bridge is not `READY`

I don't agree with the idea of using a DTO internally. I see no reason why we can't use our `Bridge` entity for Services within the Fleet Manager.

I would also prefer to keep the `and p.bridge.customerId=:customerId` on our queries for Processors out of paranoia.

@lucamolteni @vaibhavjainwiz WDYT?

Reply:
@robpblake As you suggested, I had made changes into my PR. But I am not sure with marking `@Transactional` over `ProcessorsAPI`.  Is there be any race condition scenerio if both entities loaded in different transaction?